cg/meta/report/api.py

```diff
@@ -43,16 +46,97 @@ def __init__(
         self.scout = scout_api
         self.report_validator = ReportValidator(store)
 
-    def create_delivery_report(self, case_id: str, accept_missing_data: bool = False) -> str:
+    def create_delivery_report(
+        self, case_id: str, analysis_date: datetime, accept_missing_data: bool = False
+    ) -> str:
         """Generate the html contents of a delivery report."""
 
-        delivery_data = self._get_delivery_data(case_id)
+        delivery_data = self._get_delivery_data(case_id, analysis_date)
         self._handle_missing_report_data(accept_missing_data, delivery_data, case_id)
         report_data = self._make_data_presentable(delivery_data)
         self._handle_missing_report_data(accept_missing_data, report_data, case_id)
         rendered_report = self._render_delivery_report(report_data)
         return rendered_report
 
+    def create_delivery_report_file(
+        self,
+        case_id: str,
+        file_path: Path,
+        analysis_date: datetime,
+        accept_missing_data: bool = False,
+    ) -> TextIO:
+        """Generate a temporary file containing a delivery report."""
+
+        delivery_report = self.create_delivery_report(
+            case_id=case_id, accept_missing_data=accept_missing_data, analysis_date=analysis_date
+        )
+        file_path.mkdir(parents=True, exist_ok=True)
+        report_file_path = Path(file_path / "delivery-report.html")
+        delivery_report_file = open(report_file_path, "w")
+        delivery_report_file.write(delivery_report)
+        delivery_report_file.close()
+
+        return delivery_report_file
+
+    @staticmethod
+    def add_delivery_report_to_hk(
+        delivery_report_file: Path,
+        hk_api: HousekeeperAPI,
+        case_id: str,
+        analysis_date: datetime,
+    ) -> Optional[housekeeper.store.models.File]:
+        """
+        Add a delivery report to a analysis bundle for a case in Housekeeper
+
+        If there is already a delivery report for the bundle doesn't add to it
+        If successful the method returns a pointer to the new file in Housekeeper
+        """
+        delivery_report_tag_name = "delivery-report"
+
+        version_obj = hk_api.version(case_id, analysis_date)
+
+        uploaded_delivery_report_files = hk_api.get_files(
+            bundle=case_id,
+            tags=[delivery_report_tag_name],
+            version=version_obj.id,
+        )
+        number_of_delivery_reports = len(uploaded_delivery_report_files.all())
+        is_bundle_missing_delivery_report = number_of_delivery_reports == 0
```

Comment:
You have created a function below that does exactly this, why not use that function?
```suggestion
        is_bundle_missing_delivery_report = False
        try:
            uploaded_delivery_report_file = self.get_delivery_report_from_hk(
                hk_api=hk_api,
                family_id=case_id
            )
        except FileNotFoundError:
            is_bundle_missing_delivery_report = True
```